io-engine/src/bdev/nvmx/handle.rs

```diff
@@ -632,20 +631,20 @@ impl BlockDeviceHandle for NvmeDeviceHandle {
 
         inner.account_io();
         let ret = match r.await.expect("Failed awaiting at read_at()") {
-            NvmeStatus::Generic(GenericStatusCode::Success) => {
+            NvmeStatus::Success => {
                 inner.get_io_stats_controller().account_block_io(
                     IoType::Read,
                     1,
                     num_blocks,
                 );
                 Ok(buffer.len())
             }
-            NvmeStatus::MediaError(
-                MediaErrorStatusCode::DeallocatedOrUnwrittenBlock,
-            ) => Err(CoreError::ReadingUnallocatedBlock {
-                offset,
-                len: buffer.len(),
-            }),
+            NvmeStatus::UnwrittenBlock => {
```

Comment:
Can you confirm, we've just changed the enum's here right, we haven't actually changed the error codes on the wire?

Reply:
Yes, correct, it's just enum names changed. Rust wrapper for NMVE error codes wasn't not particularly well written, so I refactored it to refer to enums from SPDK headers directly. And I added shorthands for some commonly used code, to reduce code clutter.